samcli/commands/deploy/command.py

```diff
@@ -154,6 +154,8 @@
     required=False,
     is_flag=True,
     help="Automatically resolve s3 bucket for non-guided deployments."
+    "Enabling this option will also create a managed default s3 bucket for you."
+    "If you do not provide a --s3-bucket value, the managed bucket will be used."
```

Comment:
Add a space at the end of the string. Otherwise the output will be 
`Enabling this option will also create a managed default s3 bucket for you.(missing space)If you do not provide a --s3-bucket value, the managed bucket will be used.`


```
    "Enabling this option will also create a managed default s3 bucket for you. "
    "If you do not provide a --s3-bucket value, the managed bucket will be used. "
```